src/validation.cpp

```diff
@@ -2022,18 +2022,24 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
         minterKey = pindex->minterKey();
         auto nodeId = mnview.GetMasternodeIdByOperator(minterKey);
         assert(nodeId);
-        auto const & node = *mnview.GetMasternode(*nodeId);
-        if (node.mintedBlocks + 1 != block.mintedBlocks)
+        auto nodePtr = mnview.GetMasternode(*nodeId);
+        if (!nodePtr) {
+            // NOTE(sp): Add branch prediction optimization here?
+            return state.Invalid(ValidationInvalidReason::CONSENSUS, error("ConnectBlock(): Failed to find masternode %s in the database",
+                                                                           nodeId->ToString()), REJECT_INVALID, "bad-masternode-id");
+        }
```

Comment:
Just assert, if data is missing that means storage corruption.

Reply:
`\#if defined(NDEBUG)
\# error "Defi cannot be compiled without assertions."
\#endif`

Usually, release builds switch off assertions. but here we can use it. will update.